cl/sentinel/handlers/handlers.go

```diff
@@ -73,6 +101,57 @@ func NewConsensusHandlers(ctx context.Context, db persistence.RawBeaconBlockChai
 	return c
 }
 
+func (limiter *RateLimiter) incrementCounter() {
+	atomic.AddUint32(&limiter.counter, 1)
+}
+
+func (limiter *RateLimiter) resetCounter() {
+	atomic.StoreUint32(&limiter.counter, 0)
+}
+
+func (limiter *RateLimiter) loadCounter() uint32 {
+	return atomic.LoadUint32(&limiter.counter)
+}
+
+func (c *ConsensusHandlers) checkRateLimit(peerId string, method string, limit uint32) error {
+	// Initialize
+	if _, ok := c.peerRateLimits[peerId]; !ok {
+		c.peerRateLimits[peerId] = make(map[string]*RateLimiter)
+	}
+	if _, ok := c.peerRateLimits[peerId][method]; !ok {
+		c.peerRateLimits[peerId][method] = &RateLimiter{counter: 0, lastRequestTime: time.Now(), limited: false}
+	}
+
+	limiter := c.peerRateLimits[peerId][method]
+
+	// Check if the peer is in the punishment period
+	now := time.Now()
+	if limiter.limited {
+		punishmentEndTime := limiter.lastRequestTime.Add(punishmentPeriod * time.Minute)
```

Comment:
just do `.Add(punishmentPeriod)` and move the `* time.Minute` up

Reply:
Fixed